src/js/tooltip.js

```diff
@@ -20,17 +20,47 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
      * A grade component that is to be attached on components where tooltips are needed
      *
      * To use it: provide the option "tooltipContentMap" that defines the mapping between
-     * the names in the selectors block and message bundle for dom elements to have tooltips:
+     * the names in the selectors block and message bundle for dom elements to have tooltips.
+     * Two methods to define mappings:
+     *
+     * 1. The direct mapping between one element and one label
      * {
      *     "back": "backLabel"
      * }
      * The left hand side is the name in the selectors block for the element to have the tooltip.
      * The right hand side is the name in the message bundle for the content to be shown for that element.
+     *
+     * 2. The mapping btw one common selector used by multiple DOM elements and their correspondig tooltips.
+     * This is typically used by defining tooltips for rendered radio buttons or checkboxes. When
+     * "langRow": {
+     *     tooltip: ["label1", "label2"...],
+     *     tooltipAtSelect: ["label1-at-select", "label2-at-select"...]
+     * }
+     * The left hand side is same as method 1 to define the name in the selectors block for elements to have tooltips.
+     * The right hand side is an object with two paths: tooltip & tooltipAtSelect. Their values are arrays. "tooltip"
+     * array contains names in the message bundle to be shown when mapped elements are not selected. "tooltipAtSelect"
+     * contains names when elements are selected. The number of elements in these arrays should be same as the number
+     * of elements that the left hand selector is attached to.
      */
     fluid.defaults("gpii.firstDiscovery.attachTooltip", {
         gradeNames: ["fluid.viewRelayComponent", "autoInit"],
         tooltipOptions: {},
         tooltipContentMap: {},  // Must be provided by integrators
+        model: {
+            // The index of the currently selected element
+            currentSelectedIndex: 0  // Must be updated by integrators.
+        },
+        modelListeners: {
+            currentSelectedIndex: {
+                listener: "{that}.tooltip.applier.change",
```

Comment:
We should be able to replace this with a "change record" rather than a manual invocation of applier.change - http://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#example-of-declarative-triggering-of-changes